### DISCUSSION TITLE
Remix Installation Instructions

### DISCUSSION THREAD

**Discussion by joshmedeski** [tailwindlabs/tailwindcss#6293] [Ideas]
I would love to see [installation documentation](https://tailwindcss.com/docs/installation) for how to get started with [Remix](https://remix.run).

**Comment by NovusOpus** [comment]
#### Probably going to be included in the Tailwind CSS V3 Docs.

#### In the mean time you can follow this steps:

#### 1. After Installing Remix, install this two dev dependencies:
```console
npm add -D concurrently tailwindcss
```
#### 2. Initialize a `tailwind.config file` with
```console
npx tailwindcss init -p
```
#### 3. Update your `tailwind.config` file so we can tell it which files to generate classes from:

For `Tailwind V2`:
```diff
module.exports = {
+  mode: "jit",
+  purge: ["./app/**/*.{ts,tsx}"],
  theme: {
    extend: {}
  },
  variants: {},
  plugins: []
};
```
For `Tailwind V3`:
```diff
module.exports = {
+  content: ["./app/**/*.{ts,tsx}"],
  theme: {
    extend: {}
  },
  variants: {},
  plugins: []
};
```
#### 4. Update the package scripts in the `package.json` file to generate the tailwind file during dev and for the production build:
```diff
{
  
  "scripts": {
+    "build": "npm run build:css && remix build",
+    "build:css": "tailwindcss -o ./app/tailwind.css",
+    "dev": "concurrently \"npm run dev:css\" \"remix dev\"",
+    "dev:css": "tailwindcss -o ./app/tailwind.css --watch",
    "postinstall": "remix setup node",
    "start": "remix-serve build"
  }

}
```
#### 5.Finally, import the generated CSS file into your app `root.tsx`:
```diff
+ import styles from "./tailwind.css";

+ export function links() {
+   return [{ rel: "stylesheet", href: styles }];
+ }
```

#### If you want to use Tailwind's `@apply` method to extract custom classes, create a css file in the root directory, for example: `./styles/tailwind.css`:
```css
@tailwind base;
@tailwind components;
@tailwind utilities;

@layer components {
  .custom-class {
    @apply ...;
  }
}
```
#### Then alter how tailwind is generating css:
```diff
{
  
  "scripts": {
+    "build": "npm run build:css && remix build",
+    "build:css": "tailwindcss -i ./styles/tailwind.css -o ./app/tailwind.css",
+    "dev": "concurrently \"npm run dev:css\" \"remix dev\"",
+    "dev:css": "tailwindcss -i ./styles/tailwind.css -o ./app/tailwind.css --watch",
    "postinstall": "remix setup node",
    "start": "remix-serve build"
  }
 
}
```

## Source: [Remix Docs](https://remix.run/docs/en/v1/guides/styling#tailwind) Shared by @adamwathan in his [tweet](https://twitter.com/adamwathan/status/1467196301238542340)
